client/plots/sampleView.js

```diff
@@ -112,35 +99,84 @@ class SampleView {
 				}
 				this.app.dispatch({ type: 'plot_edit', id: this.id, config: { samples } })
 			})
-			if (config.samples.length > 15)
-				this.app.dispatch({
-					type: 'plot_edit',
-					id: this.id,
-					config: { samples: config.samples.filter((s, i) => i < 15) }
-				})
 		} else {
-			this.sampleId2Name = await this.app.vocabApi.getAllSamples()
-			const samples = Object.entries(this.sampleId2Name)
-			this.sample = config.sample || { sampleId: samples[0][0], sampleName: samples[0][1] }
-			this.dom.select
-				.selectAll('option')
-				.data(samples)
-				.enter()
-				.append('option')
-				.attr('value', d => d[0])
-				.property('selected', d => d[0] == this.sample.sampleId)
-				.html((d, i) => d[1])
-			this.dom.select.on('change', e => {
-				const sampleId = this.dom.select.node().value
-				const sampleName = this.sampleId2Name[sampleId]
-				this.sample = { sampleId, sampleName }
-				this.app.dispatch({ type: 'plot_edit', id: this.id, config: { sample: this.sample } })
+			const sampleName2Id = await this.app.vocabApi.getAllSamplesByName()
+			const allSamples = Object.keys(sampleName2Id)
+			const sampleName = allSamples[0]
+			this.sample = config.sample || { sampleId: sampleName2Id[sampleName], sampleName }
+			const input = this.dom.sampleDiv
+				.append('input')
+				.attr('list', 'sampleDatalist')
+				.attr('placeholder', sampleName)
+				.style('width', '400px')
+				.text(sampleName)
+
+			const datalist = this.dom.sampleDiv.append('datalist').attr('id', 'sampleDatalist')
+
+			input.on('keyup', e => {
+				const limit = 100
```

Comment:
Ah, I was wondering why I am only seeing a 100 samples on typing in. It works nicely. I guess my concern would be how would a user know which samples to look for? I typed in 'PA' and only 100 samples show up even though there's more than 100 samples with 'PA'.

Reply:
I increased the limit to 1000, there will be cases for large datasets where this would not list all the samples, but otherwise the rendering breaks.

Reply:
Well I went ahead and set the limit to 10000, so it would list always all the possible samples for all our datasets, except the BALL-scrna, as the datasets have less than this number of samples.